### PR TITLE
Made label retrieval faster and fixed bug

### PR DESCRIPTION
**Pull request by ghislainfourny**
Fixes https://github.com/28msec/cellstore/issues/1066

Now 900ms (was 1.2 s)

http://secxbrl-dev10.28.io/v1/_queries/public/api/facts.jq?token=foobar&tag=DOW30&concept=us-gaap:Assets&labels=true&format=html&fiscalPeriod=FY&fiscalYear=2014

### CODE REVIEW

**Comment by dknochen** [comment]
@ghislainfourny could you explain this change to me offline?

**Comment by ghislainfourny** [comment]
That's an easy question. Yes! :-)

File: queries/public/api/facts.jq [hunk 1]
```diff
@@ -36,6 +44,19 @@ declare variable $local:additional-concepts as object* := (
   }
 );
 
+declare function local:resolve-aspect-name-labels(
```
Comment (dknochen) [inline queries/public/api/facts.jq]: comments :)

**Comment by dknochen** [comment]
we could add this as a test http://secxbrl.28.io/v1/_queries/public/api/facts.jq?token=foobar&tag=DOW30&concept=us-gaap:Assets&labels=true&format=html&fiscalPeriod=FY&fiscalYear=2014
checking that all fields of legal entity are filled in

**Comment by dknochen** [comment]
@ghislainfourny performance improvements plus bug fix is always the best :+1:

**Comment by ghislainfourny** [comment]
@dknochen I added the comment and the new test!

File: queries/public/test/secxbrl/facts.jq [hunk 2]
```diff
@@ -182,6 +182,18 @@ local:check({
...
+          "facts",
+          {
+            tag : "DOW30",
+            concept:"us-gaap:Assets",
+            labels: "true",
+            fiscalPeriod: "FY",
+            fiscalYear: "2014"
+          },
+          function($res as item*) as item* { count(distinct-values($res.FactTable[].Labels."dei:LegalEntityAxis")) },
+          1
```
Comment (dknochen) [inline queries/public/test/secxbrl/facts.jq]: This wouldn't have failed even before. I think, we should really count that its 30.

Well, it looks like its 31... Something seems to be fishy with the DOW30 tag?

**Comment by dknochen** [comment]
@ghislainfourny wow, very cool resolution :)

**Comment by ghislainfourny** [comment]
Thanks ;-)

**Pull request merged**